# Project 3: SQL Data Analysis

In [ ]:
import pandas as pd

In [ ]:
import sqlite3

In [ ]:
df = pd.read_excel("/content/Cleaned_Ecommerce_Dataset.xlsx")

df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice,CalculatedTotal
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04,2504.04


In [ ]:
conn = sqlite3.connect("ecommerce.db")

In [ ]:
df.to_sql("orders", conn, if_exists = "replace", index = False)

1200

## Displaying the First Five Records

In [ ]:
query = """ select * from orders limit 5 """
pd.read_sql(query, conn)

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice,CalculatedTotal
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04,2504.04


## Filtering Data Using the WHERE Clause


In [ ]:
query = """ select OrderID, Product, PaymentMethod, TotalPrice from orders where PaymentMethod = 'Credit Card' """
pd.read_sql(query, conn)

,OrderID,Product,PaymentMethod,TotalPrice
0,ORD200002,Tablet,Credit Card,2753.40
1,ORD200005,Phone,Credit Card,491.72
2,ORD200009,Desk,Credit Card,2037.52
3,ORD200010,Tablet,Credit Card,3129.85
4,ORD200013,Laptop,Credit Card,604.47
...,...,...,...,...
229,ORD201174,Printer,Credit Card,667.02
230,ORD201178,Monitor,Credit Card,70.24
231,ORD201179,Laptop,Credit Card,2592.75
232,ORD201190,Desk,Credit Card,1849.17


## Sorting Records Using ORDER BY

In [ ]:
query = """ select Product, TotalPrice from orders order by TotalPrice desc """
pd.read_sql(query, conn)

,Product,TotalPrice
0,Tablet,3456.40
1,Monitor,3390.95
2,Laptop,3390.80
3,Chair,3384.90
4,Tablet,3370.20
...,...,...
1195,Laptop,18.20
1196,Tablet,17.98
1197,Tablet,17.24
1198,Phone,14.06


## Using Aggregate Functions

In [ ]:
query = """ select count(*) as Total_Orders from orders """
pd.read_sql(query, conn)

,Total_Orders
0,1200


In [ ]:
query = """ select sum(TotalPrice) as Total_Revenue from orders """
pd.read_sql(query, conn)

,Total_Revenue
0,1264761.96


In [ ]:
query = """ select avg(TotalPrice) as Average_Order_Value from orders """
pd.read_sql(query, conn)

,Average_Order_Value
0,1053.9683


## Revenue Analysis by Product Using GROUP BY

In [ ]:
query = """ select Product, sum(TotalPrice) as Total_Revenue from orders group by Product """
pd.read_sql(query, conn)

,Product,Total_Revenue
0,Chair,195620.11
1,Desk,167459.93
2,Laptop,192126.56
3,Monitor,175651.41
4,Phone,151722.39
5,Printer,195612.61
6,Tablet,186568.95


## Ranking Products by Revenue

In [ ]:
query = """ select Product, sum(TotalPrice) as Total_Revenue from orders group by Product order by Total_Revenue desc """
pd.read_sql(query, conn)

,Product,Total_Revenue
0,Chair,195620.11
1,Printer,195612.61
2,Laptop,192126.56
3,Tablet,186568.95
4,Monitor,175651.41
5,Desk,167459.93
6,Phone,151722.39


## Filtering High-Value Delivered Orders

In [ ]:
query = """ select OrderID, Product, TotalPrice, OrderStatus from orders where TotalPrice>1000 and OrderStatus = 'Delivered' """
pd.read_sql(query, conn)

,OrderID,Product,TotalPrice,OrderStatus
0,ORD200004,Printer,2504.04,Delivered
1,ORD200021,Monitor,1371.80,Delivered
2,ORD200029,Desk,1270.59,Delivered
3,ORD200032,Tablet,2683.60,Delivered
4,ORD200075,Tablet,1194.51,Delivered
...,...,...,...,...
86,ORD201116,Printer,1062.98,Delivered
87,ORD201145,Desk,1714.59,Delivered
88,ORD201155,Printer,2138.28,Delivered
89,ORD201164,Tablet,1577.85,Delivered


## Using HAVING to Filter Grouped Results

In [ ]:
query = """ select Product, sum(TotalPrice) as Total_Revenue from orders group by Product having Total_Revenue>50000 """
pd.read_sql(query, conn)

,Product,Total_Revenue
0,Chair,195620.11
1,Desk,167459.93
2,Laptop,192126.56
3,Monitor,175651.41
4,Phone,151722.39
5,Printer,195612.61
6,Tablet,186568.95


# Key Observations and Insights

* The dataset consists of 1,200 customer orders with a total revenue of approximately ₹1.26 million and an average order value of ₹1,053.97.

* Chair was the highest revenue-generating product, contributing approximately ₹195,620.11, followed closely by Printer with revenue of ₹195,612.61.

* Online Payment was the most frequently used payment method, accounting for 258 orders, indicating a strong preference for digital transactions.

* The order status distribution was relatively balanced, with Cancelled orders (250) slightly higher than Returned orders (247).

* Revenue analysis revealed that Chair, Printer, and Laptop were among the top-performing products, making significant contributions to overall sales.

* SQL queries involving COUNT(), SUM(), AVG(), GROUP BY, and HAVING were used to extract and summarize meaningful business insights from the dataset.
